packages/backend/src/clients/cube/transformer.ts

```diff
@@ -49,23 +74,34 @@ export const cubeTransfomers: SemanticLayerTransformer<
     SqlQuery
 > = {
     fieldsToSemanticLayerFields: (dimensions, metrics) => {
-        const semanticDimensions: SemanticLayerField[] = dimensions.map(
-            (d) => ({
+        const semanticDimensions: SemanticLayerField[] = dimensions.map((d) => {
+            const type = getSemanticLayerTypeFromCubeType(d.type);
+
+            // TODO: check if cube has a function to get available granularities
+            const availableGranularities =
+                type === SemanticLayerFieldType.TIME
+                    ? Object.values(granularityMap)
+                    : [];
```

Comment:
In the docs I haven't been able to find if there's a way to get the available granularities, so I am defaulting to allowing all